### ISSUE TITLE
[telemetry] Instrument AjaxInvoker with spans and metrics

### ISSUE THREAD

**Issue by phax** [phax/ph-oton#40]
## Goal

Instrument AJAX function invocation in `ph-oton-ajax` with a ph-telemetry `SERVER` span plus counters and a duration histogram.

Together with the REST API invoker this covers the two programmatic request entry points of a ph-oton application.

## Where

`ph-oton-ajax/src/main/java/com/helger/photon/ajax/AjaxInvoker.java`

```java
public void invokeFunction (@NonNull final String sFunctionName,
                            @NonNull final IAjaxExecutor aAjaxExecutor,
                            @NonNull final IRequestWebScopeWithoutResponse aRequestScope,
                            @NonNull final PhotonUnifiedResponse aAjaxResponse) throws Exception
```

The method already declares `throws Exception`, so `Telemetry.withSpanVoidThrowing (..., Exception)` fits without any signature change.

It already has a `StopWatch`, a `try` / `catch (Exception)` and three `StatisticsManager` handlers. **Keep the statistics handlers as they are** and add telemetry alongside them.

## Restructuring note

Unlike `APIInvoker`, this method has **no `finally` block** — the `StopWatch` is stopped at the end of the `try`, so a failing invocation currently records no time at all. When adding the histogram, move the duration measurement into a `finally` block so failures are timed too. That is a behaviour improvement worth calling out in the news entry.

## Cardinality

`sFunctionName` is registered up-front in the `IAjaxRegistry` — it is a bounded set and is safe as a metric attribute. This is unlike the REST API case, where the concrete request path must not be used.

## What to emit

New `com.helger.photon.ajax.CAjaxTelemetry`:

| Constant | Value |
|---|---|
| `SPAN_INVOKE` | `photon.ajax.invoke` |
| `METRIC_INVOCATIONS` | `photon.ajax.invocations` |
| `METRIC_DURATION` | `photon.ajax.duration` |
| `ATTR_AJAX_FUNCTION` | `photon.ajax.function` |
| `ATTR_AJAX_SUCCESS` | `photon.ajax.success` |
| `UNIT_REQUEST` | `{request}` |
| `UNIT_MILLIS` | `ms` |

New `com.helger.photon.ajax.AjaxMetrics` (counter + histogram, `public static final`, created at class load) and a package-private `com.helger.photon.ajax.AjaxTelemetry` emitter, both mirroring `LongRunningJobMetrics` / `LongRunningJobTelemetry`.

**Span** `photon.ajax.invoke`, kind `SERVER`, wrapping the whole method body, attribute `photon.ajax.function`. Exceptions are recorded and re-thrown by `withSpanVoidThrowing`.

**Counter** `photon.ajax.invocations`, attributes `function` + `success`, incremented on every path through the method.

**Histogram** `photon.ajax.duration` in ms, attributes `function` + `success`, recorded from the new `finally` block.

## Long-running executions

`AjaxSettings.getLongRunningExecutionLimitTime ()` already defines a threshold and fires `longRunningExecutionCallbacks`. Consider adding `aSpan.addEvent (...)` when that threshold is exceeded so slow calls are visible in the trace, not just in the callback. Optional — decide during implementation.

## POM

`ph-oton-ajax/pom.xml` needs a new dependency (no `<version>` — the BOM is imported in the root POM):

```xml
<dependency>
  <groupId>com.helger.telemetry</groupId>
  <artifactId>ph-telemetry</artifactId>
</dependency>
```

## Acceptance criteria

- [ ] Span, counter and histogram emitted as specified.
- [ ] Duration is now recorded for failing invocations too (moved into `finally`).
- [ ] Existing `StatisticsManager` handlers untouched.
- [ ] Unit test with recording SPIs installed via `Telemetry.install (...)` / `TelemetryMetrics.install (...)` and reset to `null` in `@After`, covering a successful and a failing invocation. Model it on `ph-oton-mgrs/src/test/java/.../longrun/LongRunningJobTelemetryTest.java`.
- [ ] `mvn install` passes; `README.md` news bullet added.

---

### Shared conventions for all ph-telemetry work in ph-oton

**Facades.** `com.helger.telemetry.Telemetry` for spans, `com.helger.telemetry.TelemetryMetrics` for instruments. If no `ITelemetryTracerSPI` / `ITelemetryMeterSPI` is registered by the deploying application, everything degrades to cheap no-ops — so emit unconditionally, never guard with an "is telemetry enabled" flag.

**No OpenTelemetry in ph-oton.** Only `com.helger.telemetry:ph-telemetry` is ever added. `ph-telemetry-otel` and the OpenTelemetry SDK are the deploying application's business. The version comes from the `ph-telemetry-parent-pom` BOM already imported in the root `pom.xml`, so the new `<dependency>` needs no `<version>`.

**Reference implementation.** `ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/longrun/` — `CLongRunningJobTelemetry` (names), `LongRunningJobMetrics` (instrument holder), `LongRunningJobTelemetry` (package-private emitter). Follow that layout.

**Constants: one class per module** (decided; `CLongRunningJobTelemetry` stays where it is for backwards compatibility since it is public API since 10.4.0):

| Module | Constants class |
|---|---|
| ph-oton-io | `com.helger.photon.io.CIOTelemetry` |
| ph-oton-app | `com.helger.photon.app.CAppTelemetry` |
| ph-oton-audit | `com.helger.photon.audit.CAuditTelemetry` |
| ph-oton-ajax | `com.helger.photon.ajax.CAjaxTelemetry` |
| ph-oton-api | `com.helger.photon.api.CAPITelemetry` |
| ph-oton-security | `com.helger.photon.security.CSecurityTelemetry` |
| ph-oton-mgrs | `CLongRunningJobTelemetry` (existing) + `com.helger.photon.mgrs.CMgrsTelemetry` (new) |
| ph-oton-core | `com.helger.photon.core.CCoreTelemetry` |
| ph-oton-connect | `com.helger.photon.connect.CConnectTelemetry` |
| ph-oton-uicore | `com.helger.photon.uicore.CUICoreTelemetry` |
| ph-oton-datatables | `com.helger.photon.uictrls.datatables.CDataTablesTelemetry` |

Each is `@Immutable`, `final`, with a private constructor, and groups constants under `// === span names ===`, `// === metric instrument names ===`, `// === attribute keys ===`, `// === metric units ===` comments, exactly as `CLongRunningJobTelemetry` does. Every constant gets a Javadoc comment.

**Naming.** Spans and metrics `photon.<area>.<thing>`, attributes `photon.<area>.<key>`. Units follow the OpenTelemetry convention: `ms` for durations, `{request}` / `{login}` / `{job}` / `{session}` for counts.

**Cardinality is a hard rule.** Metric attributes must be bounded and small. Anything unbounded — user IDs, session IDs, concrete URL paths, execution IDs, IP addresses, exception messages — goes on **spans only**, never on a metric. This mirrors `ATTR_JOB_TYPE` (metric) vs `ATTR_JOB_EXECUTION_ID` (span-only) in `CLongRunningJobTelemetry`.

**Span/thread binding.** A span is bound to the thread that created it (see the warning block at the end of `LongRunningJobTelemetry.onJobEnd`). Prefer `Telemetry.withSpan*` so start and end are guaranteed to happen on the same thread; only fall back to a manual `startSpan` + try/finally when the span genuinely has to outlive the method.

**Existing `StatisticsManager` handlers stay.** Where a site already emits ph-commons statistics, telemetry is added *alongside* them — do not remove or change the statistics handlers. The long-term strategy for the two systems is tracked separately.

**Coding style.** phax styleguide: Hungarian notation, `final` parameters, JSpecify `@NonNull`/`@Nullable`, `_` prefix for private methods, private methods declared before their callers, space before `(` and `<`, inline string concatenation in log calls (no SLF4J `{}` placeholders), Apache 2.0 header.

**Definition of done.** `mvn install` passes; a bullet is added to the `v10.5.0 - work in progress` section of `README.md` under "News and noteworthy".

**Comment by phax** [comment]
Part of #55 — see that issue for the shared conventions, the module-level constants class table and the POM impact overview.

**Comment by phax** [comment]
Implemented for **10.5.1**.

### What was added

| File | Kind |
|---|---|
| `ph-oton-ajax/.../ajax/CAjaxTelemetry.java` | new, public - span / metric / attribute names |
| `ph-oton-ajax/.../ajax/AjaxMetrics.java` | new, public - counter `photon.ajax.invocations`, histogram `photon.ajax.duration` |
| `ph-oton-ajax/.../ajax/AjaxTelemetry.java` | new, package-private emitter |
| `ph-oton-ajax/.../ajax/AjaxInvoker.java` | modified |
| `ph-oton-ajax/pom.xml` | `com.helger.telemetry:ph-telemetry` added (no `<version>`, BOM) |
| `ph-oton-ajax/src/test/.../AjaxTelemetryTest.java` | new, 2 tests |

### Emission

`invokeFunction (...)` is wrapped in `Telemetry.<Exception> withSpanVoidThrowing (CAjaxTelemetry.SPAN_INVOKE, ETelemetrySpanKind.SERVER, ...)`. No signature change was needed.

* Span `photon.ajax.invoke` (`SERVER`) with `photon.ajax.function` and `photon.ajax.success`.
* Counter `photon.ajax.invocations` and histogram `photon.ajax.duration` with `function` + `success`, emitted from the new `finally` block on every code path.

`sFunctionName` comes from the `IAjaxRegistry` and is a bounded set, so it is used as a metric attribute - unlike the concrete REST path in #39.

### Restructuring (behaviour change)

As called out in the issue, the `StopWatch` was created *inside* the `try` and stopped at the end of it, so a failing invocation was never timed. The stopwatch is now created before the `try` and stopped in a new `finally` block.

Consequence beyond telemetry - deliberate, and now documented in the news entry:

* the statistics timer `AjaxInvoker$timer` is now also fed for **failing** invocations, and
* the registered `IAjaxLongRunningExecutionCallback`s are now also triggered for failing invocations.

Both bring `AjaxInvoker` in line with `APIInvoker`, which has always done exactly this in its `finally` block.

### Long running executions - the optional span event

Decided **against** it. Because the long-running threshold check now lives in the `finally` block, the span is already closed at that point, and `ITelemetrySpan` explicitly specifies all mutators as no-ops after `close ()`. Moving the check back inside the span would mean measuring the duration twice or leaking the stopwatch into the lambda. The histogram `photon.ajax.duration` exposes the same information per invocation, so nothing is lost.

### Not touched

All three `StatisticsManager` handlers and the callback invocations are unchanged.

### Verification

Full reactor `mvn install` on all 23 modules: BUILD SUCCESS, no test failures. `AjaxTelemetryTest`: 2/2 green, including the assertion that the duration histogram is recorded with `photon.ajax.success=false` for a failing invocation. `README.md` news bullet added under the new `v10.5.1 - work in progress` section.
